src/ng/directive/form.js

```diff
@@ -131,6 +128,28 @@ function FormController(element, attrs, $scope, $animate, $interpolate) {
     if (control.$name) {
       form[control.$name] = control;
     }
+
+    control.$$parentForm = form;
+
+    forEach(control.$pending, function(value, name) {
```

Comment:
this may not be a big deal in most cases, but 3x `forEach()` calls seems like a perf hit, which would be most noticeable in forms with visibility controlled by ng-if or ng-switch. I'm not saying it's a huge problem, but yeah